test/src/code_field/split_highlight_lines_test.dart

```diff
@@ -0,0 +1,67 @@
+import 'package:flutter_code_editor/src/highlight/result.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:highlight/highlight.dart';
+import 'package:highlight/languages/dart.dart';
+
+const examples = {
+  //
+  'Empty': '',
+
+  'Single line comment': '''
+public class MyClass {
+  public void main() { //comment
+  }
+}''',
+
+  'Readonly comment': '''
+public class MyClass {
+  public void main() { // readonly
+  }
+}''',
+
+  'Named sections': '''
+class MyClass {
+  void readOnlyMethod() {// [START section1]
+  }// [END section1]
+  // [START section2]
+  void method() {
+  }// [END section2]
+}''',
+
+  'Invalid code (Folded)': '''
+class MyClass {
+  void readOnlyMethod() {
+
+  void method() {
+}''',
+};
+
+void main() {
+  group('Highlight split lines test.', () {
+    examples.forEach((name, text) {
+      test(name, () {
+        Result? highlighted;
+
+        final mode = dart;
+        highlight.registerLanguage('language', mode);
+
+        highlighted = highlight.parse(text, language: 'language');
+
+        expect(
+          highlighted.splitLines().nodes!.any((node) => node.hasSeveralLines()),
+          false,
```

Comment:
This test is insufficient. It can succeed even if `splitLines()` just deletes everything.
We must check that no extra splits are there and nothing is lost.